homeassistant/components/aprilaire/coordinator.py

```diff
@@ -0,0 +1,213 @@
+        self.client.stop_listen()
+
+    async def wait_for_ready(
+        self, ready_callback: Callable[[bool], Awaitable[bool]]
+    ) -> bool:
+        """Wait for the client to be ready."""
+
+        if not self.data or Attribute.MAC_ADDRESS not in self.data:
+            data = await self.client.wait_for_response(
+                FunctionalDomain.IDENTIFICATION, 2, WAIT_TIMEOUT
+            )
+
+            if not data or Attribute.MAC_ADDRESS not in data:
+                _LOGGER.error("Missing MAC address")
+                await ready_callback(False)
+
+                return False
+
+        if not self.data or Attribute.NAME not in self.data:
+            await self.client.wait_for_response(
+                FunctionalDomain.IDENTIFICATION, 4, WAIT_TIMEOUT
+            )
+
+        if not self.data or Attribute.THERMOSTAT_MODES not in self.data:
+            await self.client.wait_for_response(
+                FunctionalDomain.CONTROL, 7, WAIT_TIMEOUT
+            )
+
+        if (
+            not self.data
+            or Attribute.INDOOR_TEMPERATURE_CONTROLLING_SENSOR_STATUS not in self.data
+        ):
+            await self.client.wait_for_response(
+                FunctionalDomain.SENSORS, 2, WAIT_TIMEOUT
+            )
+
+        await ready_callback(True)
+
+        return True
+
+    @property
+    def device_name(self) -> str:
+        """Get the name of the thermostat."""
+
+        return self.create_device_name(self.data)
+
+    def create_device_name(self, data: Optional[dict[str, Any]]) -> str:
+        """Create the name of the thermostat."""
+
+        name = data.get(Attribute.NAME) if data else None
+
+        return name if name else "Aprilaire"
+
+    def get_hw_version(self, data: dict[str, Any]) -> str:
+        """Get the hardware version."""
+
+        if hardware_revision := data.get(Attribute.HARDWARE_REVISION):
+            return (
+                f"Rev. {chr(hardware_revision)}"
+                if hardware_revision > ord("A")
+                else str(hardware_revision)
+            )
+
+        return "Unknown"
+
+    @property
+    def device_info(self) -> DeviceInfo | None:
+        """Get the device info for the thermostat."""
+        return self.create_device_info(self.data)
+
+    def create_device_info(self, data: dict[str, Any]) -> DeviceInfo | None:
+        """Create the device info for the thermostat."""
+
+        if data is None or Attribute.MAC_ADDRESS not in data or self.unique_id is None:
+            return None
+
+        device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.unique_id)},
+            name=self.create_device_name(data),
+            manufacturer="Aprilaire",
+        )
+
+        model_number = data.get(Attribute.MODEL_NUMBER)
+        if model_number is not None:
+            device_info["model"] = (
+                MODELS[model_number]
+                if model_number in MODELS
+                else f"Unknown ({model_number})"
```

Comment:
resolve Ruff complaint
```suggestion
                MODELS.get(model_number, f"Unknown ({model_number})")
```